zivid_camera/test/test_zivid_camera.cpp

```diff
@@ -718,493 +1013,140 @@ TEST_F(ZividNodeTest, testCaptureCameraInfo)
-    ASSERT_TRUE(boost::filesystem::exists(cmd.request.file_path));
-    ASSERT_FALSE(ros::service::call(serviceName, cmd));
-  }
-
-  template <typename CmdType>
-  void testLoadNonExistentFileGivesError(const std::string & serviceName)
-  {
-    ASSERT_TRUE(ros::service::waitForService(serviceName, short_wait_duration));
-    CmdType cmd;
-    cmd.request.file_path = "/tmp/foo/bar";
-    ASSERT_FALSE(boost::filesystem::exists(cmd.request.file_path));
-    ASSERT_FALSE(ros::service::call(serviceName, cmd));
-  }
-
-private:
-  template <typename Node>
-  static void recursivelyFillInUnsetWithCameraDefault(
-    Node & node, const Zivid::CameraInfo & cameraInfo)
-  {
-    if constexpr (
-      Node::nodeType == Zivid::DataModel::NodeType::group ||
-      Node::nodeType == Zivid::DataModel::NodeType::leafDataModelList) {
-      node.forEach([&cameraInfo](auto & child) {
-        recursivelyFillInUnsetWithCameraDefault(child, cameraInfo);
-      });
-    } else if (!node.hasValue()) {
-      static_assert(Node::nodeType == Zivid::DataModel::NodeType::leafValue);
-      node = Zivid::Experimental::SettingsInfo::defaultValue<Node>(cameraInfo);
+      auto response = doSrvRequest<zivid_interfaces::srv::CaptureAndSave>(
+        capture_and_save_service_name, request, std::chrono::seconds{10});
+      ASSERT_TRUE(response);
+      verifyTriggerResponseError(response);
+      ASSERT_FALSE(std::filesystem::exists(file_path));
     }
+    executor_.spin_some();
+    all_capture_topics_subscriber.assert_num_topics_received(1);
   }
 };
 
-TEST_F(LoadSettingsTest, testLoadSettingsFromFile)
-{
-  testLoadSettingsFromFile<zivid_camera::LoadSettingsFromFile, Zivid::Settings>(
-    load_settings_from_file_service_name,
-    {"3d/single.yml", "3d/hdr.yml", "3d/hdr_with_not_set_values.yml", "3d/single.yml"});
-}
+TEST_F(CaptureAndSaveTest, testCaptureAndSaveEmptyPathProvided) { captureAndSaveToPath("", false); }
 
-TEST_F(LoadSettingsTest, testLoadSettings2DFromFile)
+TEST_F(CaptureAndSaveTest, testCaptureAndSaveInvalidPath)
 {
-  testLoadSettingsFromFile<zivid_camera::LoadSettings2DFromFile, Zivid::Settings2D>(
-    load_settings_2d_from_file_service_name,
-    {"2d/single_1.yml", "2d/single_2.yml", "2d/single_with_not_set_values.yml", "2d/single_1.yml"});
+  captureAndSaveToPath("invalid_path", false);
 }
 
-TEST_F(LoadSettingsTest, testLoadSettingsFromInvalidFileGivesError)
+TEST_F(CaptureAndSaveTest, testCaptureAndSaveInvalidExtension)
 {
-  testLoadInvalidSettingsGivesError<zivid_camera::LoadSettingsFromFile>(
-    load_settings_from_file_service_name);
+  captureAndSaveToPath("/tmp/invalid_extension.wrong", false);
 }
 
-TEST_F(LoadSettingsTest, testLoadSettings2DFromInvalidFileGivesError)
-{
-  testLoadInvalidSettingsGivesError<zivid_camera::LoadSettings2DFromFile>(
-    load_settings_2d_from_file_service_name);
-}
+TEST_F(CaptureAndSaveTest, testCaptureAndSaveZDF) { captureAndSaveToPath("/tmp/valid.zdf", true); }
```

Comment:
I think we should extract the temporary file code from `TmpFile` to get a platform-independent temporary path. Applies to all `/tmp` paths.

Reply:
168d43bdd1ce3b1c131ff6b49e12365868f821eb